File: backend/models/Bill.js
const mongoose = require('mongoose');

const billSchema = new mongoose.Schema(
  {
    tripId: { type: mongoose.Schema.Types.ObjectId, ref: 'Trip', required: true },
    billDate: { type: Date, required: true },
    tripDate: { type: Date, required: true },
    vehicleNumber: { type: String, required: true },
    tripDetails: { type: String, required: true },
    startTime: { type: Date },
    endTime: { type: Date },
    startKm: { type: Number, required: true },
    endKm: { type: Number, required: true },
    ratePerKm: { type: Number, required: true, min: 0 },
    dayRent: { type: Number, default: 0 },
    hourRent: { type: Number, default: 0 },
    numberOfDays: { type: Number, default: 0 },
    numberOfHours: { type: Number, default: 0 },
    driverBata: { type: Number, default: 0 },
    tollCharges: { type: Number, default: 0 },
    permitCharges: { type: Number, default: 0 },
    parkingCharges: { type: Number, default: 0 },
    advanceReceived: { type: Number, default: 0 },
    totalKm: { type: Number, required: true },
    kmCharge: { type: Number, required: true },
    totalAmount: { type: Number, required: true },
    payableAmount: { type: Number, required: true },
    pdfPath: { type: String },
    paymentStatus: { type: String, enum: ['paid', 'pending'], default: 'pending' },
  },
  { timestamps: true }
);

module.exports = mongoose.model('Bill', billSchema);

File: flutter_app/lib/models/bill.dart
class BillModel {
  final String id;
  final String vehicleNumber;
  final double totalAmount;
  final double payableAmount;
  final String paymentStatus;

  const BillModel({
    required this.id,
    required this.vehicleNumber,
    required this.totalAmount,
    required this.payableAmount,
    required this.paymentStatus,
  });

  factory BillModel.fromJson(Map<String, dynamic> json) {
    return BillModel(
      id: json['_id'] as String,
      vehicleNumber: json['vehicleNumber'] as String,
      totalAmount: (json['totalAmount'] as num).toDouble(),
      payableAmount: (json['payableAmount'] as num).toDouble(),
      paymentStatus: json['paymentStatus'] as String? ?? 'pending',
    );
  }
}
